### ISSUE TITLE
Heading levels in ZUIHeader don't accurately represent the semantic structure of the page

### ISSUE THREAD

**Issue by henrycatalinismith** [zetkin/app.zetkin.org#3259]
## Description

[WCAG success criterion 1.3.1](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html) is about info and relationships, with the idea being that the page's semantic structure should be programmatically determinable for assistive technology. Heading levels are an important navigational cue for users of screen readers who may otherwise be unable to perceive font sizes and weights.

ZUIHeader is used at the top of both SimpleLayout and TabbedLayout to render the topmost page heading. The string passed as the `title` prop to ZUIHeader is most appropriately rendered as an `<h1>` element but is currently rendered as a `<div>`.  The string passed as the `subtitle` prop is most appropriately rendered as a `<div>` or `<p>` element but is currently rendered as a `<h2>`.

## Steps to reproduce

1. Load https://app.dev.zetkin.org/organize/1/projects.
2. Start a screen reader.
3. Inspect the page's headings.

### Expected outcome

All Projects & Activities should be the top-level heading. There should not be an `<h2>` element underneath it, both because in this case no `subtitle` prop exists and because if one did it would not represent a deeper level in the semantic hierarchy of the page but rather extra text accopanying the title.

### Actual outcome

All Projects & Activities is absent from the headings list as it's rendered as a `<div>`. An empty `<h2>` element appears in the list of headings.

## Screenshots

![Image](https://github.com/user-attachments/assets/3197b11b-33ba-4d17-8900-ceb20e0b3892)
